File: 0948-sort-an-array/0948-sort-an-array.java
class Solution {
    public int[] sortArray(int[] nums) {
        quickSort(nums,0,nums.length-1);
        return nums;

    }
    static void  quickSort(int[] arr,int low,int hi) {
		if(low>=hi) {
			return;
		}
	int s=low;
	int e=hi;
	int n=s+(e-s)/2;
	int pivot=arr[n];
	
	while(s<=e) {
		while(arr[s]<pivot) {
			s++;
		}
		while(arr[e]>pivot) {
			e--;
		}
		if(s<=e) {
			int temp=arr[s];
			arr[s]=arr[e];
			arr[e]=temp;
			s++;
			e--;
		}
		 
	}
	 quickSort(arr,low,e);
	 quickSort(arr,s,hi);
	}
}

File: 0948-sort-an-array/0948-sort-an-array.py
class Solution:
    def sortArray(self, nums):
        self.mergeSort(nums, 0, len(nums))
        return nums

    def mergeSort(self, arr, s, e):
        if e - s <= 1:
            return

        m = s + (e - s) // 2

        self.mergeSort(arr, s, m)
        self.mergeSort(arr, m, e)

        self.merge(arr, s, m, e)

    def merge(self, arr, s, m, e):
        i, j, k = s, m, 0
        mix = [0] * (e - s)

        while i < m and j < e:
            if arr[i] < arr[j]:
                mix[k] = arr[i]
                i += 1
            else:
                mix[k] = arr[j]
                j += 1
            k += 1

        while i < m:
            mix[k] = arr[i]
            i += 1
            k += 1
        while j < e:
            mix[k] = arr[j]
            j += 1
            k += 1

        for l in range(len(mix)):
            arr[s + l] = mix[l]